raiden/network/proxies/token_network.py

```diff
@@ -1566,175 +1567,184 @@ def update_transfer_light(
-                raise RaidenRecoverableError(msg)
 
-            if detail.settle_block_number < failed_at_blocknumber:
-                raise RaidenRecoverableError(
-                    "update_transfer transation sent after settlement window"
+                detail = self._detail_channel(
+                    participant1=self.node_address,
+                    participant2=partner,
+                    block_identifier=failed_at_blockhash,
+                    channel_identifier=channel_identifier,
                 )
 
-            # At this point it is known the channel is CLOSED on block
-            # `failed_at_blockhash`
-            partner_details = self._detail_participant(
-                channel_identifier=channel_identifier,
-                participant=partner,
-                partner=self.node_address,
-                block_identifier=failed_at_blockhash,
-            )
+                if detail.state < ChannelState.CLOSED:
+                    msg = (
+                        f"cannot call update_transfer channel has not been closed yet. "
+                        f"current_state={detail.state}"
+                    )
+                    raise RaidenUnrecoverableError(msg)
 
-            if not partner_details.is_closer:
-                raise RaidenUnrecoverableError(
-                    "update_transfer cannot be sent if the partner did not close the channel"
+                if detail.state >= ChannelState.SETTLED:
+                    msg = (
+                        f"cannot call update_transfer channel has been settled already. "
+                        f"current_state={detail.state}"
+                    )
+                    raise RaidenRecoverableError(msg)
+
+                if detail.settle_block_number < failed_at_blocknumber:
+                    raise RaidenRecoverableError(
+                        "update_transfer transation sent after settlement window"
+                    )
+
+                # At this point it is known the channel is CLOSED on block
+                # `failed_at_blockhash`
+                partner_details = self._detail_participant(
+                    channel_identifier=channel_identifier,
+                    participant=partner,
+                    partner=self.node_address,
+                    block_identifier=failed_at_blockhash,
                 )
 
-            raise RaidenUnrecoverableError("update_transfer failed for an unknown reason")
+                if not partner_details.is_closer:
+                    raise RaidenUnrecoverableError(
+                        "update_transfer cannot be sent if the partner did not close the channel"
+                    )
 
-        log.info("updateNonClosingBalanceProof successful", **log_details)
+                raise RaidenUnrecoverableError("update_transfer failed for an unknown reason")
+        except Exception:
+            log.info("An error occurred trying to update the non closing balance proof for channel id " +
```

Comment:
Why dont we just catch the `RaidenUnrecoverableError `and `RaidenRecoverableError` here? 

I think is more natural to make other errors just fail at runtime, any other exception that is not one of those two should not be an expected exception

Reply:
Because the first call is to the smart contract, if anything else happens that throws an error then you will need to repeat the action, so suppose that everything is fine with the smart contract call but later something fails then you will force to call the smart contract again, so i wanted to avoid that but it's your call, if you want we can only catch those kind of errors

Reply:
Thatas a good reason.

Still seems too wide for me to catch any exception, but since we dont want to loose gas innecesary and i dont have any better option in mind right now, we can go with this implementation.

Reply:
Ok cool, i mean i was thinking on the gas basically so i can't think in another way to avoid spending gas in this case.